src/util/alloc/immix_allocator.rs

```diff
@@ -229,7 +221,19 @@ impl<VM: VMBinding> ImmixAllocator<VM> {
     fn alloc_slow_hot(&mut self, size: usize, align: usize, offset: isize) -> Address {
         trace!("{:?}: alloc_slow_hot", self.tls);
         if self.acquire_recyclable_lines(size, align, offset) {
-            self.alloc(size, align, offset)
+            // If stress test is active, then we need to go to the slow path instead of directly
+            // calling alloc(). This is because the `acquire_recyclable_lines()` function
+            // manipulates the cursor and limit; if we directly call `alloc()` then we will miss
+            // updating the allocation as the TLAB will service the allocation request. If we set
+            // the limit in `acquire_recyclable_lines()`, then we risk running into an infinite
+            // loop of failing the fastpath and then trying to allocate from a recyclable line.
+            let stress_test = self.plan.base().is_stress_test_gc_enabled();
+            let precise_stress = self.plan.base().is_precise_stress();
+            if stress_test && precise_stress {
```

Comment:
This seems to be the only change that may affect performance. It would be better to have it measured.

Reply:
Alright -- will measure it :+1: